src/components/CodeEditor.machine.ts

```diff
@@ -0,0 +1,81 @@
+import { Machine } from 'xstate';
+import * as actions from './CodeEditor.actions';
+import * as services from './CodeEditor.services';
+import {
+  CodeEditorContext,
+  CodeEditorSchema,
+  CodeEditorEvents,
+  EditorContext,
+} from './CodeEditor.types';
+
+export default Machine<CodeEditorContext, CodeEditorSchema, CodeEditorEvents>(
+  {
+    id: 'editor',
+    initial: 'editor',
+    context: {
+      external: {
+        selectedRequest: undefined,
+        requestMetaDataById: undefined,
+      },
```

Comment:
I can see this data is also available in our outer context, from Core. Is this the XState's way of reusing values from an outer context? By duplicating them? So, whenever our outer context structure changes, do we need to change our components' `useEffect`'s dependency list to also update our inner context?

I remember I had this problem in the past too and I got this question - that's why I'm curious. :joy:

Reply:
Being honest, I'm not sure 😂 I couldn't find anything in the documentation as well. I'm starting thinking better about actors and it might be the case here. But I need more time to study and test it, I'm gonna try to implement something about it and send a PR later if it works well.

Reply:
Oh, don't worry - we're on the same boat! Let's let the time teach us then :joy: